File: Array/1431-Kids-With-the-Greatest-Number-of-Candies/1431-Kids-With-the-Greatest-Number-of-Candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        List<Boolean> ans = new ArrayList<>();

        int maxNum = candies[0];

        for(int i = 1;i<candies.length;i++) {
            maxNum = Math.max(maxNum, candies[i]);
        }

        for(int candy : candies) {
            if(candy + extraCandies >= maxNum){
                ans.add(true);
            }else{
                ans.add(false);
            }
        }

        return ans;
    }
}

File: Array/1431-Kids-With-the-Greatest-Number-of-Candies/1431-Kids-With-the-Greatest-Number-of-Candies.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        vector<bool> ans;
        int maxNum = 0;

        for(int candy : candies){
            maxNum = max(candy, maxNum);
        }

        for(int candy : candies){
            if(candy + extraCandies >= maxNum) {
                ans.push_back(true);
            } else {
                ans.push_back(false);
            }
        }
        return ans;
    }
};
